homeassistant/components/mqtt/device_tracker.py

```diff
@@ -13,22 +13,35 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_PAYLOAD_CUSTOM = "payload_custom"
```

Comment:
Instead of like this, we have this config option generally like this: `payload_home` and `payload_not_home`. We should do that here too.

Reply:
I was originally going to use that approach, but doesn't the device tracker component support other defined zones as well? How would you suggest I handle translating payloads to other zones?

I could add all three parameters if you see the value in having support for both.

Reply:
Here's what I came up with in order to support both the home/not_home zones as well as custom zones. If you don't want to support custom zones then I can remove `payload_custom` and the corresponding logic entirely:

```python
_LOGGER = logging.getLogger(__name__)

CONF_PAYLOAD_CUSTOM = "payload_custom"
CONF_PAYLOAD_HOME = "payload_home"
CONF_PAYLOAD_NOT_HOME = "payload_not_home"

DEFAULT_PAYLOAD_HOME = "home"
DEFAULT_PAYLOAD_NOT_HOME = "not_home"

PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(mqtt.SCHEMA_BASE).extend(
    {
        vol.Required(CONF_DEVICES): {cv.string: mqtt.valid_subscribe_topic},
        vol.Optional(CONF_PAYLOAD_CUSTOM): {cv.string: cv.string},
        vol.Optional(CONF_PAYLOAD_HOME, default=DEFAULT_PAYLOAD_HOME): cv.string,
        vol.Optional(
            CONF_PAYLOAD_NOT_HOME, default=DEFAULT_PAYLOAD_NOT_HOME
        ): cv.string,
    }
)


async def async_setup_scanner(hass, config, async_see, discovery_info=None):
    """Set up the MQTT tracker."""
    devices = config[CONF_DEVICES]
    qos = config[CONF_QOS]
    payload_custom = config.get(CONF_PAYLOAD_CUSTOM)
    payload_home = config[CONF_PAYLOAD_HOME]
    payload_not_home = config[CONF_PAYLOAD_NOT_HOME]

    for dev_id, topic in devices.items():

        @callback
        def async_message_received(msg, dev_id=dev_id):
            """Handle received MQTT message."""
            if msg.payload == payload_home:
                location_name = DEFAULT_PAYLOAD_HOME
            elif msg.payload == payload_not_home:
                location_name = DEFAULT_PAYLOAD_NOT_HOME
            elif payload_custom and msg.payload in payload_custom:
                location_name = payload_custom[msg.payload]
            else:
                location_name = msg.payload

            hass.async_create_task(
                async_see(dev_id=dev_id, location_name=location_name)
            )

        await mqtt.async_subscribe(hass, topic, async_message_received, qos)
```

Reply:
Zones in device tracker are defined by provided GPS coordinates, which are then matched against zones defined in Home Assistant. Let's stick to just home and not home for now.

Reply:
Understood